core/embed/rust/src/ui/component/base.rs

```diff
@@ -208,28 +208,48 @@ where
 /// Same as `Child` but also handles screen clearing when layout is first
 /// painted.
 pub struct Root<T> {
-    inner: Child<T>,
+    inner: Option<Child<T>>,
     marked_for_clear: bool,
 }
 
 impl<T> Root<T> {
     pub fn new(component: T) -> Self {
         Self {
-            inner: Child::new(component),
+            inner: Some(Child::new(component)),
             marked_for_clear: true,
         }
     }
 
+    pub fn inner_mut(&mut self) -> &mut Child<T> {
+        if let Some(ref mut c) = self.inner {
+            c
+        } else {
+            fatal_error!("deallocated", "Root object is deallocated")
+        }
+    }
+
     pub fn inner(&self) -> &Child<T> {
-        &self.inner
+        if let Some(ref c) = self.inner {
+            c
+        } else {
+            fatal_error!("deallocated", "Root object is deallocated")
+        }
     }
 
     pub fn skip_paint(&mut self) {
-        self.inner.skip_paint()
+        self.inner_mut().skip_paint();
     }
 
     pub fn clear_screen(&mut self) {
-        self.marked_for_clear = true;
```

Comment:
nit: i'm thinking, don't bother checking `self.inner` here if you don't need it? the requirement is "you must not corrupt memory if someone calls a function on the deleted object", we don't care if someone calls `clear()` on it and it does nothing.

Reply:
fixed https://github.com/trezor/trezor-firmware/commit/26a1d1695336fcf6b2d7cf082cc8852292987ed3